soh/soh/Enhancements/custom-message/CustomMessageInterfaceAddon.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+const char* Interface_ReplaceSpecialCharacters(char text[]);
+#ifdef __cplusplus
+};
+#endif
```

Comment:
It would take me a bit too long to flesh out the ideas in question and figure out if they are good or not, and I don't want to make this PR wait for that. What I'm thinking would involve either a C-compatible API for CustomMessages or moving a lot of the rendering code for Boss Rush to C++.

This is perfectly fine as is for a first iteration, just seeing it sparked some inspiration in me.